### PR TITLE
Add vanilla TF example

### PR DESCRIPTION
**Pull request by kshitij12345**
# Description

Include a summary of the changes and the related issue.

__Related to:__ https://app.clickup.com/t/31vb8yq

---

## ❔ This change

- [x] adds a new feature
- [ ] fixes breaking code
- [ ] is cosmetic (refactoring/reformatting)

---

## ✔️ Pre-merge checklist

Mark which items have been completed:

- [x] Refactored code ([sourcery](https://sourcery.ai/))
- [x] Tested code locally
- [x] Added code to GitHub tests ([notebooks](workflows/test-notebooks.yml), [scripts](workflows/test-scripts.yml))
- [ ] Updated GitHub [README](../README.md)

### 🪝 Run the following to both [pre-commit](https://pre-commit.com/) hooks and JupyterLab (turn on `format on save`)

- [x] Stripped outputs from notebooks - ([nbstripout](https://pypi.org/project/nbstripout/))
- [x] Sorted imports - ([isort](https://pycqa.github.io/isort/))
- [x] Formatted code - ([back](https://github.com/psf/black))

### 📄 Update documentation

Coordinate with @normandy7

- [ ] Update content in the [docs](https://docs.neptune.ai)
- [ ] (For new content) Add links to examples [README](../README.md)

---

## 🧪 Test Configuration

- OS: Ubuntu
- Python version: 3.8.12
- Neptune version: 0.16.14
- Affected libraries with version: tensorflow==2.9.2, numpy==1.21.6, requests==2.23.0

---

## ✔️ Post-merge checklist

These tasks should be done after the code is merged to main. Create a subtask under the main ClickUp task to have these on your radar and share the subtask here.

__Post-merge subtask:__ https://app.clickup.com/t/32xngcd , https://app.clickup.com/t/32xngem

- [ ] Update relevant blogs with impacted code/functionality (Contact @Patrycja-J for the list of blogs)
- [ ] Add functionality under the Neptune column of the `competitor comparison sheet` (Coordinate with @SiddhantSadangi)
- [ ] Update integration README (Contact @Patrycja-J)

### CODE REVIEW

**Comment by kshitij12345** [comment]
NOTE: Relevant TF version is not available on Python 3.11

File: integrations-and-supported-tools/tensorflow/scripts/requirements.txt [hunk 1]
```diff
@@ -0,0 +1,4 @@
+neptune-client
+tensorflow==2.9.2
+numpy==1.21.6
```
Comment (SiddhantSadangi) [inline integrations-and-supported-tools/tensorflow/scripts/requirements.txt:3]:
```suggestion
numpy
```

File: integrations-and-supported-tools/tensorflow/scripts/requirements.txt [hunk 2]
```diff
@@ -0,0 +1,4 @@
...
+requests==2.23.0
```
Comment (SiddhantSadangi) [inline integrations-and-supported-tools/tensorflow/scripts/requirements.txt:4]:
```suggestion
requests
```

File: integrations-and-supported-tools/tensorflow/scripts/run_examples.sh [hunk 3]
```diff
@@ -0,0 +1,5 @@
+echo "Installing requirements..."
+pip install -r requirements.txt
```
Comment (SiddhantSadangi) [inline integrations-and-supported-tools/tensorflow/scripts/run_examples.sh:2]:
```suggestion
pip install -U -r requirements.txt
```

**Review by SiddhantSadangi — CHANGES_REQUESTED** [review]
@kshitij12345 
Can you run the notebook through black? I see a few single quotes `'` in the code
Can you also enable source tracking (git, source code)?

File: integrations-and-supported-tools/tensorflow/notebooks/Neptune_Tensorflow.ipynb [hunk 4]
```diff
@@ -0,0 +1,496 @@
...
+    "\n",
+    "In this example, we will use Neptune to log meta-data generated from training using Tensorflow.\n",
+    "\n",
+    "By the end of this guide, you will be able to\n",
+    "* Track and version the data.\n",
+    "* Log losses and other metrics generated from training.\n",
+    "* Log prediction over multiple epochs.\n",
+    "* Save the generated model with model registry.\n",
+    "\n",
+    "[See this example in Neptune](https://app.neptune.ai/common/tensorflow-support/e/TFSUP-12)"
```
Comment (SiddhantSadangi) [inline integrations-and-supported-tools/tensorflow/notebooks/Neptune_Tensorflow.ipynb:22]:
```suggestion
    "[See this example in Neptune](https://app.neptune.ai/common/tensorflow-support/e/TFSUP-101)"
```

File: integrations-and-supported-tools/tensorflow/notebooks/Neptune_Tensorflow.ipynb [hunk 5]
```diff
@@ -0,0 +1,496 @@
...
+   "cell_type": "markdown",
+   "metadata": {
+    "tags": [
+     "comment"
+    ]
+   },
+   "source": [
+    "## Explore the results in Neptune\n",
+    "\n",
+    "You can also check out an [example run](https://app.neptune.ai/common/tensorflow-support/e/TFSUP-12) and the corresponding [model version](https://app.neptune.ai/common/tensorflow-support/m/TFSUP-TFMOD/v/TFSUP-TFMOD-25/metadata)"
```
Comment (SiddhantSadangi) [inline integrations-and-supported-tools/tensorflow/notebooks/Neptune_Tensorflow.ipynb:470]:
```suggestion
    "You can also check out an [example run](https://app.neptune.ai/common/tensorflow-support/e/TFSUP-101) and the corresponding [model version](https://app.neptune.ai/common/tensorflow-support/m/TFSUP-TFMOD/v/TFSUP-TFMOD-25/metadata)"
```

**Review by SiddhantSadangi — CHANGES_REQUESTED** [review]
Updated `showcase-run`

File: integrations-and-supported-tools/tensorflow/notebooks/Neptune_Tensorflow.ipynb [hunk 6]
```diff
@@ -0,0 +1,496 @@
...
+    "model_version['datasets/version'].track_files('mnist.npz')"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Saves model artifacts to `weights folder`\n",
```
Comment (normandy7) [inline integrations-and-supported-tools/tensorflow/notebooks/Neptune_Tensorflow.ipynb:434]:
```suggestion
    "# Saves model artifacts to "weights" folder\n",
```

File: integrations-and-supported-tools/tensorflow/notebooks/Neptune_Tensorflow.ipynb [hunk 7]
```diff
@@ -0,0 +1,496 @@
...
+    "                                                                epoch_accuracy.result()))"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Tracking model with Neptune Model Registry\n",
+    "\n",
+    "Refer the [documentation](https://neptune.ai/product/model-registry) for more information"
```
Comment (normandy7) [inline integrations-and-supported-tools/tensorflow/notebooks/Neptune_Tensorflow.ipynb:398]:
```suggestion
    "Refer to the [documentation](https://neptune.ai/product/model-registry) for more information."
```

File: integrations-and-supported-tools/tensorflow/scripts/Neptune_Tensorflow.py [hunk 8]
```diff
@@ -0,0 +1,153 @@
...
+    api_token=neptune.ANONYMOUS_API_TOKEN,
+)
+
+# (Neptune) Log meta-data to model version
+model_version["run_id"] = run["sys/id"]
+model_version["metrics/test_loss"] = test_loss
+model_version["metrics/test_accuracy"] = test_acc
+model_version["datasets/version"].track_files("mnist.npz")
+
+# Saves model artifacts to `weights folder`
```
Comment (normandy7) [inline integrations-and-supported-tools/tensorflow/scripts/Neptune_Tensorflow.py:150]:
```suggestion
# Saves model artifacts to "weights" folder
```

File: integrations-and-supported-tools/tensorflow/scripts/Neptune_Tensorflow.py [hunk 9]
```diff
@@ -0,0 +1,153 @@
...
+
+    if epoch % 5 == 0 or epoch == (params["num_epochs"] - 1):
+        print(
+            "Epoch {:03d}: Loss: {:.3f}, Accuracy: {:.3%}".format(
+                epoch, epoch_loss_avg.result(), epoch_accuracy.result()
+            )
+        )
+
+# Tracking model with Neptune Model Registry
+# Refer the [documentation](https://neptune.ai/product/model-registry) for more information
```
Comment (normandy7) [inline integrations-and-supported-tools/tensorflow/scripts/Neptune_Tensorflow.py:135]:
```suggestion
# Refer to the [documentation](https://neptune.ai/product/model-registry) for more information
```

**Review by normandy7 — CHANGES_REQUESTED** [review]

**Comment by kshitij12345** [comment]
@SiddhantSadangi I was thinking of updating the README once 

* this PR is merged so that we can test and correctly add the colab link and github link.
* the doc page is ready so that we have a link to the doc page.

Wdyt?

**Comment by kshitij12345** [comment]
@SiddhantSadangi @normandy7 have addressed the review. PTAL :)

**Comment by SiddhantSadangi** [comment]
> @SiddhantSadangi I was thinking of updating the README once
> 
>     * this PR is merged so that we can test and correctly add the colab link and github link.
links follow the same format as existing scripts/notebooks. just have to ensure that the folder and file names are correct

>     * the doc page is ready so that we have a link to the doc page.
@normandy7 takes care of this once the docs are published

**Review by SiddhantSadangi — CHANGES_REQUESTED** [review]
LGTG once the readme has been updated.
We generally update this as a part of the changes so there's no lag.

**Comment by kshitij12345** [comment]
@SiddhantSadangi have updated the README. PTAL :)

File: README.md [hunk 10]
```diff
@@ -95,6 +95,7 @@ You can run every example with zero setup as an "ANONYMOUS" Neptune user (no reg
...
 | 🤗 Transformers | [![docs]](https://docs.neptune.ai/integrations-and-supported-tools/model-training/hugging-face) | [![neptune]](https://app.neptune.ai/o/showcase/org/project-text-summarization-hf/e/PROJ-138/dashboard/Custom-Dashboard-97370bc5-ee32-48ce-a630-fc806a370e13)  | [![github]](https://github.com/neptune-ai/examples/blob/main/integrations-and-supported-tools/transformers/scripts) | [![colab]](https://colab.research.google.com/github/neptune-ai/examples/blob/master/integrations-and-supported-tools/transformers/notebooks/Neptune_Transformers.ipynb) |
+| TensorFlow | [![docs]](https://docs.neptune.ai/integrations-and-supported-tools/model-training/tensorflow) | [![neptune]](https://app.neptune.ai/common/tensorflow-support/e/TFSUP-101) | [![github]](https://github.com/neptune-ai/examples/blob/main/integrations-and-supported-tools/tensorflow/scripts) | [![colab]](https://colab.research.google.com/github/neptune-ai/examples/blob/master/integrations-and-supported-tools/tensorflow/notebooks/Neptune_Tensorflow.ipynb) |
```
Comment (SiddhantSadangi) [inline README.md:98]:
```suggestion
| TensorFlow | [![docs]](https://docs.neptune.ai/integrations-and-supported-tools/model-training/tensorflow) | [![neptune]](https://app.neptune.ai/o/common/org/tensorflow-support/e/TFSUP-101/dashboard/Overview-97f6ac04-2c4b-4d10-97da-cd3a51bbeec8) | [![github]](https://github.com/neptune-ai/examples/blob/main/integrations-and-supported-tools/tensorflow/scripts) | [![colab]](https://colab.research.google.com/github/neptune-ai/examples/blob/master/integrations-and-supported-tools/tensorflow/notebooks/Neptune_Tensorflow.ipynb) |
```

**Review by SiddhantSadangi — CHANGES_REQUESTED** [review]

**Review by SiddhantSadangi — APPROVED** [review]

File: README.md [hunk 11]
```diff
@@ -95,6 +95,7 @@ You can run every example with zero setup as an "ANONYMOUS" Neptune user (no reg
...
 | 🤗 Transformers | [![docs]](https://docs.neptune.ai/integrations-and-supported-tools/model-training/hugging-face) | [![neptune]](https://app.neptune.ai/o/showcase/org/project-text-summarization-hf/e/PROJ-138/dashboard/Custom-Dashboard-97370bc5-ee32-48ce-a630-fc806a370e13)  | [![github]](https://github.com/neptune-ai/examples/blob/main/integrations-and-supported-tools/transformers/scripts) | [![colab]](https://colab.research.google.com/github/neptune-ai/examples/blob/master/integrations-and-supported-tools/transformers/notebooks/Neptune_Transformers.ipynb) |
+| TensorFlow | [![docs]](https://docs.neptune.ai/integrations-and-supported-tools/model-training/tensorflow) | [![neptune]](https://app.neptune.ai/o/common/org/tensorflow-support/e/TFSUP-101/dashboard/Overview-97f6ac04-2c4b-4d10-97da-cd3a51bbeec8) | [![github]](https://github.com/neptune-ai/examples/blob/main/integrations-and-supported-tools/tensorflow/scripts) | [![colab]](https://colab.research.google.com/github/neptune-ai/examples/blob/master/integrations-and-supported-tools/tensorflow/notebooks/Neptune_Tensorflow.ipynb) |
```
Comment (normandy7) [inline README.md:98]:
```suggestion
| TensorFlow | [![docs]](https://docs.neptune.ai/integrations/tensorflow) | [![neptune]](https://app.neptune.ai/o/common/org/tensorflow-support/e/TFSUP-101/dashboard/Overview-97f6ac04-2c4b-4d10-97da-cd3a51bbeec8) | [![github]](https://github.com/neptune-ai/examples/blob/main/integrations-and-supported-tools/tensorflow/scripts) | [![colab]](https://colab.research.google.com/github/neptune-ai/examples/blob/master/integrations-and-supported-tools/tensorflow/notebooks/Neptune_Tensorflow.ipynb) |
```

File: integrations-and-supported-tools/tensorflow/notebooks/Neptune_Tensorflow.ipynb [hunk 12]
```diff
@@ -0,0 +1,514 @@
...
+    ]
+   },
+   "source": [
+    "## Start a run\n",
+    "\n",
+    "To connect your script to Neptune and create a new run, we tell Neptune:\n",
+    "* **Who you are** - with a Neptune API token\n",
+    "* **Where to send your data** - to a Neptune project\n",
+    "\n",
+    "The cell below lets you record data to the public project [common/quickstarts](https://app.neptune.ai/common/quickstarts) as an anonymous user."
```
Comment (normandy7) [inline integrations-and-supported-tools/tensorflow/notebooks/Neptune_Tensorflow.ipynb:69]:
```suggestion
    "The cell below lets you record data to the public project [common/tensorflow-support](https://app.neptune.ai/common/tensorflow-support) as an anonymous user."
```

File: integrations-and-supported-tools/tensorflow/notebooks/Neptune_Tensorflow.ipynb [hunk 13]
```diff
@@ -0,0 +1,514 @@
...
+    "                epoch, epoch_loss_avg.result(), epoch_accuracy.result()\n",
+    "            )\n",
+    "        )"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Tracking model with Neptune Model Registry\n",
```
Comment (normandy7) [inline integrations-and-supported-tools/tensorflow/notebooks/Neptune_Tensorflow.ipynb:409]:
```suggestion
    "Tracking model with Neptune model registry\n",
```

File: integrations-and-supported-tools/tensorflow/scripts/Neptune_Tensorflow.py [hunk 14]
```diff
@@ -0,0 +1,153 @@
...
+        )
+
+    if epoch % 5 == 0 or epoch == (params["num_epochs"] - 1):
+        print(
+            "Epoch {:03d}: Loss: {:.3f}, Accuracy: {:.3%}".format(
+                epoch, epoch_loss_avg.result(), epoch_accuracy.result()
+            )
+        )
+
+# Tracking model with Neptune Model Registry
```
Comment (normandy7) [inline integrations-and-supported-tools/tensorflow/scripts/Neptune_Tensorflow.py:134]:
```suggestion
# Tracking model with Neptune model registry
```

**Review by SiddhantSadangi — APPROVED** [review]

**Review by SiddhantSadangi — CHANGES_REQUESTED** [review]
@kshitij12345 , @normandy7 
We should also rename the `tensorflow-keras` folder to just `keras`. Links will also need to be updated in the docs

**Comment by kshitij12345** [comment]
@SiddhantSadangi I think this should be out-of-scope for this PR and can be done in a separate PR. (I am conservative as it may break anything point to it and block this PR).

**Review by SiddhantSadangi — APPROVED** [review]

**Comment by SiddhantSadangi** [comment]
> @SiddhantSadangi I think this should be out-of-scope for this PR and can be done in a separate PR. (I am conservative as it may break anything pointing to that code and block this PR).

sounds good.. we'll rename the project too with the PR

File: README.md [hunk 15]
```diff
@@ -96,6 +96,7 @@ You can run every example with zero setup as an "ANONYMOUS" Neptune user (no reg
...
 | 🤗 Transformers | [![docs]](https://docs.neptune.ai/integrations-and-supported-tools/model-training/hugging-face) | [![neptune]](https://app.neptune.ai/o/showcase/org/project-text-summarization-hf/e/PROJ-138/dashboard/Custom-Dashboard-97370bc5-ee32-48ce-a630-fc806a370e13)  | [![github]](https://github.com/neptune-ai/examples/blob/main/integrations-and-supported-tools/transformers/scripts) | [![colab]](https://colab.research.google.com/github/neptune-ai/examples/blob/master/integrations-and-supported-tools/transformers/notebooks/Neptune_Transformers.ipynb) |
+| TensorFlow | [![docs]](https://docs.neptune.ai/integrations/tensorflow) | [![neptune]](https://app.neptune.ai/o/common/org/tensorflow-support/e/TFSUP-101/dashboard/Overview-97f6ac04-2c4b-4d10-97da-cd3a51bbeec8) | [![github]](https://github.com/neptune-ai/examples/blob/main/integrations-and-supported-tools/tensorflow/scripts) | [![colab]](https://colab.research.google.com/github/neptune-ai/examples/blob/master/integrations-and-supported-tools/tensorflow/notebooks/Neptune_Tensorflow.ipynb) |
```
Comment (normandy7) [inline README.md:99]: Let's just remove the docs link for now. I won't have time to add a page until the 1st week of January.

```suggestion
| TensorFlow | | [![neptune]](https://app.neptune.ai/o/common/org/tensorflow-support/e/TFSUP-101/dashboard/Overview-97f6ac04-2c4b-4d10-97da-cd3a51bbeec8) | [![github]](https://github.com/neptune-ai/examples/blob/main/integrations-and-supported-tools/tensorflow/scripts) | [![colab]](https://colab.research.google.com/github/neptune-ai/examples/blob/master/integrations-and-supported-tools/tensorflow/notebooks/Neptune_Tensorflow.ipynb) |
```

File: integrations-and-supported-tools/tensorflow/notebooks/Neptune_Tensorflow.ipynb [hunk 16]
```diff
@@ -0,0 +1,514 @@
...
+   "cell_type": "markdown",
+   "metadata": {
+    "tags": [
+     "comment"
+    ]
+   },
+   "source": [
+    "## Explore the results in Neptune\n",
+    "\n",
+    "You can also check out an [example run](https://app.neptune.ai/common/tensorflow-support/e/TFSUP-101) and the corresponding [model version](https://app.neptune.ai/common/tensorflow-support/m/TFSUP-TFMOD/v/TFSUP-TFMOD-112/metadata)"
```
Comment (normandy7) [inline integrations-and-supported-tools/tensorflow/notebooks/Neptune_Tensorflow.ipynb:483]:
```suggestion
    "You can also check out an [example run](https://app.neptune.ai/common/tensorflow-support/e/TFSUP-101) and the corresponding [model version](https://app.neptune.ai/common/tensorflow-support/m/TFSUP-TFMOD/v/TFSUP-TFMOD-112/metadata)."
```

File: integrations-and-supported-tools/tensorflow/scripts/Neptune_Tensorflow.py [hunk 17]
```diff
@@ -0,0 +1,153 @@
...
+
+    if epoch % 5 == 0 or epoch == (params["num_epochs"] - 1):
+        print(
+            "Epoch {:03d}: Loss: {:.3f}, Accuracy: {:.3%}".format(
+                epoch, epoch_loss_avg.result(), epoch_accuracy.result()
+            )
+        )
+
+# Tracking model with Neptune model registry
+# Refer to the [documentation](https://neptune.ai/product/model-registry) for more information
```
Comment (normandy7) [inline integrations-and-supported-tools/tensorflow/scripts/Neptune_Tensorflow.py:135]: Could have been me that suggested this should be Markdown link formatted -- sorry about that. Realized now this is a script, not notebook.

```suggestion
# For more, refer to the documentation: https://neptune.ai/product/model-registry
```

**Review by normandy7 — DISMISSED** [review]
Looks good otherwise.

**Pull request merged**